### PR TITLE
Fix small docstring typos and formatting

### PR DESCRIPTION
**Pull request by ihincks**
### Summary

This is a non-comprehensive set of tiny fixes for the docstrings.

### CODE REVIEW

File: qiskit_dynamics/models/rotating_frame.py [hunk 1]
```diff
@@ -279,7 +279,7 @@ def state_out_of_frame(
         y_in_frame_basis: Optional[bool] = False,
         return_in_frame_basis: Optional[bool] = False,
     ) -> Array:
-        r"""Take a state out of the rotating frame, i.e. ``return exp(tF) @ y``.
+        r"""Take a state out of the rotating frame, i.e. return ``(tF) @ y``.
```
Comment (DanPuzzuoli) [inline qiskit_dynamics/models/rotating_frame.py:282]:
```suggestion
        r"""Take a state out of the rotating frame, i.e. return ``exp(tF) @ y``.
```

File: qiskit_dynamics/models/__init__.py [hunk 2]
```diff
@@ -107,7 +107,7 @@
 
 .. code-block:: python
 
-    model.evaluation_mode = 'sparse'
+    model.evaluation_mode = "sparse"
```
Comment (DanPuzzuoli) [inline qiskit_dynamics/models/__init__.py:110]: Is there a reason to prefer double tick here?

**Review by DanPuzzuoli — CHANGES_REQUESTED** [review]
lgtm, aside from one question and one correction to one of your changes.

**Comment by DanPuzzuoli** [comment]
@ihincks you'll also need to sign the CLA.

File: qiskit_dynamics/models/__init__.py (hunk 2 above)
Comment (ihincks) [reply qiskit_dynamics/models/__init__.py:110]: Just because it's the `black` convention.

**Comment by ihincks** [comment]
I don't know how to resolve the CLA problem...I've tried signing it a few times. I'm going to close and open the issue out of desperation.

**Pull request closed without merging**